lib/gitlab/client/pipelines.rb

```diff
@@ -38,9 +38,25 @@ def pipeline(project, id)
     #
     # @param  [Integer, String] project The ID or name of a project.
     # @param  [String] ref Reference to commit.
+    # @param  [Hash] variables Variables passed to pipelines
     # @return [Gitlab::ObjectifiedHash] The pipelines changes.
-    def create_pipeline(project, ref)
-      post("/projects/#{url_encode project}/pipeline?ref=#{ref}")
+    def create_pipeline(project, ref, variables = {})
+      body = {}
+
+      if variables.any?
+        # This mapping is necessary, cause the API expects an array with objects (with `key` and `value` keys)
+        # See: https://docs.gitlab.com/ee/api/pipelines.html#create-a-new-pipeline
+        normalized_variables = variables.map do |(key, value)|
+          { key: key, value: value }
+        end
+        body[:variables] = normalized_variables
+      end
```

Comment:
How about this?

```rb
body[:variables] = variables.each_with_object([]) do |(key, value), arr|
  arr << { key: key, value: value }
end
```

Reply:
It would work this way:
```ruby
body[:variables] = variables.map do |(key, value)|
  { key: key, value: value }
end if variables.any?
```

But I personally do not like the trailing condition in combination with multi-line statements...

Reply:
The whole statement in one-line looks reasonable:
```ruby
body[:variables] = variables.map { |(key, value)| { key: key, value: value } } if variables.any?
```

What do you think?

Reply:
> What do you think?

lgtm

Reply:
I've updated the PR! See 6a3f6e6